database_service/mock_db.go

```diff
@@ -0,0 +1,49 @@
+package database_service
+
+import (
+	"strconv"
+
+	"github.com/leechongyan/Studtor_backend/authentication_service/models"
+	"github.com/leechongyan/Studtor_backend/helpers"
+)
+
+// mock db
+// this mock db has to implement all the methods which will be used by DatabaseConnector
+
+type Mockdb struct {
+	UserCollection
+}
+
+type UserCollection map[string]models.User
+
+func (db *Mockdb) Init() {
+	db.UserCollection = make(map[string]models.User)
+}
+
+func (db Mockdb) SaveUser(user models.User) (err *helpers.RequestError) {
```

Comment:
how does this persist if the db isn't a pointer? i think this works on `map` because golang implements `map` using pointers [actually](https://blog.golang.org/maps)

Reply:
mdb := &Mockdb{}
mdb.Init()
CurrentDatabaseConnector = mdb 

I am passing into CurrentDatabaseConnector by reference

Reply:
oh, not too sure about this. i skimmed HAHA but ok will defer to you if it works!